docs/performance/tune-performance.md

```diff
@@ -331,6 +331,52 @@ cudaProviderOptions.UpdateOptions(providerOptionsDict);
 SessionOptions options = SessionOptions.MakeSessionOptionWithCudaProvider(cudaProviderOptions);  // Dispose this finally
 ```
 
+### Using CUDA Graphs in the CUDA EP
+While using the CUDA EP, ORT supports the usage of [CUDA Graphs](https://developer.nvidia.com/blog/cuda-10-features-revealed/) to remove CPU overhead associated with launching CUDA kernels sequentially. To enable the usage of CUDA Graphs, use the provider option as shown in the samples below.
+Currently, there are some constraints with regards to using the CUDA Graphs feature which are listed below:
+
+1) Usage of CUDA Graphs is limited to models with no control-flow ops in them (i.e.) models with `If`, `Loop`, and `Scan` ops in them cannot use this feature
+2) Usage of CUDA Graphs is limited to models where-in all the model ops (graph nodes) can be partitioned to the CUDA EP
+3) Multi-threaded usage is not supported currently (i.e.) `Run()` MAY NOT be invoked on the same `InferenceSession` object from multiple threads while using CUDA Graphs
+
+NOTE: The very first `Run()` performs a variety of tasks under the hood like making CUDA memory allocations, capturing the CUDA graph for the model, and then performing a graph replay to ensure that the graph runs. Due to this, the latency associated with the first `Run()` is bound to be high. The subsequent `Run()`s only perform graph replays of the graph captured and cached in the first `Run()`. 
+
+* Python
+```
+providers = [("CUDAExecutionProvider", {"enable_cuda_graph": '1'})]
+sess_options = ort.SessionOptions()
+sess = ort.InferenceSession("my_model.onnx",  sess_options = sess_options, providers=providers)
```

Comment:
Could we add some code here to show how to create `io_binding` and update the input values in place for each CUDA Graph run? The usage for CUDA Graph is a little different from the normal `Run(...)` because  the input addresses could not be changed. It will be great if we can add some explanation for these in the above document.

Reply:
Talked about IOBinding in the explanation above and updated the samples to reflect it. I realized as I was crafting the samples that the tests added by you are very succinct samples themselves. So ended up using them here :)